dds/DCPS/RTPS/Sedp.cpp

```diff
@@ -268,6 +269,50 @@ Sedp::init(const RepoId& guid, const RtpsDiscovery& disco,
   return DDS::RETCODE_OK;
 }
 
+void
+Sedp::unicast_locators(OpenDDS::DCPS::LocatorSeq& locators) const
+{
+  DCPS::RtpsUdpInst_rch rtps_inst =
+    DCPS::static_rchandle_cast<DCPS::RtpsUdpInst>(transport_inst_);
+  using namespace OpenDDS::RTPS;
+
+  CORBA::ULong idx = 0;
+
+  // multicast first so it's preferred by remote peers
+  if (rtps_inst->use_multicast_ && rtps_inst->multicast_group_address_ != ACE_INET_Addr()) {
+    idx = locators.length();
+    locators.length(idx + 1);
+    locators[idx].kind = address_to_kind(rtps_inst->multicast_group_address_);
+    locators[idx].port = rtps_inst->multicast_group_address_.get_port_number();
+    RTPS::address_to_bytes(locators[idx].address,
+      rtps_inst->multicast_group_address_);
+  }
+
+  if (rtps_inst->local_address_config_str_.empty()) {
+    typedef OPENDDS_VECTOR(ACE_INET_Addr) AddrVector;
+    AddrVector addrs;
+    OpenDDS::DCPS::get_interface_addrs(addrs);
+    AddrVector::iterator adr_it = addrs.begin();
+    while (adr_it != addrs.end()) {
+      idx = locators.length();
+      locators.length(idx + 1);
+      locators[idx].kind = address_to_kind(*adr_it);
+      locators[idx].port = rtps_inst->local_address_.get_port_number();
+      RTPS::address_to_bytes(locators[idx].address,
+        *adr_it);
+      ++adr_it;
+    }
+  }
+  else {
```

Comment:
Please check for coding style issues like this one: the two lines 305 and 306 should be combined.